### PR TITLE
Add new memcpy API

### PR DESCRIPTION
**Pull request by jnie-TT**
Add `memcpy(void *, Tensor)` API

tt-xla stores the source runtime tensor but only a dest data pointer, thus this API will support this use-case and save the user the trouble of creating a runtime tensor just to copy the data into a pointer.

FYI @kmitrovicTT

### CODE REVIEW

**Review by kmitrovicTT — APPROVED** [review]

File: runtime/lib/ttnn/runtime.cpp
Comment (jnie-TT) [reply runtime/lib/ttnn/runtime.cpp:306]: Couple reasons:
- My PR for runtime stitching was already big so I didn't want to complicate things further
- I'm not super familiar with metal runtime, and it seems `Tensor` could wrap an array of tensors for metal runtime, and I don't know how the user would want to interface with this
- The other runtime stitching APIs aren't implemented for metal and the submit API is not refactored, therefore having this API will probably have no use-case in the short-term

Therefore I decided to scope the PR to solely TTNN implementation, and we can add this later to metal runtime once we have plans on how to update the metal runtime to support something similar.

**Comment by kmabeeTT** [comment]
@jnie-TT any concerns to merge this one after approvals?  @mrakitaTT tells me tt-xla needs this in order to consume newest tt-mlir.  FYI @nsmithtt / @tapspatel for pending approval.

**Review by nsmithtt — APPROVED** [review]

**Review by tapspatel — APPROVED** [review]

**Pull request merged**
